dfusion_rust_core/src/models/solution.rs

```diff
@@ -29,29 +30,36 @@ impl Solution {
 
 impl Serializable for Solution {
     fn bytes(&self) -> Vec<u8> {
+        let mut res = (self.prices.len() as u16).to_be_bytes().to_vec();
         let alternating_buy_sell_amounts: Vec<u128> = self
             .executed_buy_amounts
             .iter()
             .zip(self.executed_sell_amounts.iter())
             .flat_map(|tup| once(tup.0).chain(once(tup.1)))
             .cloned()
             .collect();
-        [&self.prices, &alternating_buy_sell_amounts]
+
+        let prices_and_volumes: Vec<u8> = [&self.prices, &alternating_buy_sell_amounts]
             .iter()
             .flat_map(|list| list.iter())
             .flat_map(Serializable::bytes)
-            .collect()
+            .collect();
+        res.extend(prices_and_volumes);
+        res
     }
 }
 
 impl Deserializable for Solution {
     fn from_bytes(mut bytes: Vec<u8>) -> Self {
-        let volumes = bytes.split_off(TOKENS as usize * 12);
+        // First 2 bytes encode the length of price vector (i.e. num_tokens)
+        let len_prices = BigEndian::read_u16(&bytes[0..2]);
+        let volumes = bytes.split_off(2 + len_prices as usize * 12);
         let prices = bytes
+            .split_off(2)
```

Comment:
I would suggest `bytes[2..]` instead of `bytes.split_off(2)` here as the latter allocates a completely new `Vec` and since you don't re-use `bytes` it seems unnecessary.

Reply:
Will include this great nit asap... but forgot here.